### PR TITLE
mgr/dashboard: sort servers and OSDs in OSD list

### PR DESCRIPTION
**Pull request by jcsp**
Fixes: http://tracker.ceph.com/issues/21572
Signed-off-by: John Spray <john.spray@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
tested locally.

File: src/pybind/mgr/dashboard/module.py [hunk 1]
```diff
@@ -977,18 +976,18 @@ def _osds_by_server(self):
                             summary = self._osd_summary(osd_id,
                                                         osd_map.osds_by_id[osd_id])
 
-                            if first:
-                                # A little helper for rendering
-                                summary['first'] = True
-                                first = False
                             result[hostname].append(summary)
 
+                    result[hostname].sort(key=lambda a: a['id'])
```
Comment (Liuchang0812) [inline src/pybind/mgr/dashboard/module.py]: what's the type of `a['id']`? we should use `sort(key=lambda a: int(a'id'])` if it's `str`.

Comment (tchaikov) [reply src/pybind/mgr/dashboard/module.py]: please see `PyModules::dump_server()`.

Comment (Liuchang0812) [reply src/pybind/mgr/dashboard/module.py]: yeah. it's correct

**Pull request merged**